client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/RemoteQueryProtostreamAnnotationsDisableIndexingTest.java

```diff
@@ -0,0 +1,166 @@
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      org.infinispan.configuration.cache.ConfigurationBuilder builder = new org.infinispan.configuration.cache.ConfigurationBuilder();
+      builder.indexing().index(Index.ALL)
+            .addProperty("default.directory_provider", "local-heap")
+            .addProperty("lucene_version", "LUCENE_CURRENT");
+
+      return TestCacheManagerFactory.createServerModeCacheManager(builder);
+   }
+
+   @Override
+   protected RemoteCacheManager getRemoteCacheManager() {
+      org.infinispan.client.hotrod.configuration.ConfigurationBuilder clientBuilder = new org.infinispan.client.hotrod.configuration.ConfigurationBuilder();
+      clientBuilder.addServer().host("127.0.0.1").port(hotrodServer.getPort());
+      clientBuilder.marshaller(new ProtoStreamMarshaller());
+      return new RemoteCacheManager(clientBuilder.build());
+   }
+
+   @BeforeClass
+   protected void registerProtobufSchema() throws Exception {
+      //initialize client-side serialization context
+      SerializationContext serializationContext = ProtoStreamMarshaller.getSerializationContext(remoteCacheManager);
+      ProtoSchemaBuilder protoSchemaBuilder = new ProtoSchemaBuilder();
+      String protoSchemaFile = protoSchemaBuilder.fileName("memo.proto")
+            .addClass(Memo.class)
+            .addClass(Author.class)
+            .build(serializationContext);
+
+      //initialize server-side serialization context
+      RemoteCache<String, String> metadataCache = remoteCacheManager.getCache(ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME);
+      metadataCache.put("memo.proto", protoSchemaFile);
+      System.out.println("RemoteQueryProtostreamAnnotationsDisableIndexingTest.registerProtobufSchema " + protoSchemaFile);
+      assertFalse(metadataCache.containsKey(ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX));
+   }
+
+   public void testAttributeQuery() {
+      RemoteCache<Integer, Memo> remoteCache = remoteCacheManager.getCache();
+
+      remoteCache.put(1, createMemo1());
+      remoteCache.put(2, createMemo2());
+
+      // get memo1 back from remote cache and check its attributes
+      Memo fromCache = remoteCache.get(1);
+      assertMemo1(fromCache);
+
+      // get memo1 back from remote cache via query and check its attributes
+      QueryFactory qf = Search.getQueryFactory(remoteCache);
+      Query query = qf.from(Memo.class)
+            .having("text").like("%ipsum%")
+            .build();
+      List<Memo> list = query.list();
+      assertNotNull(list);
+      assertEquals(1, list.size());
+      assertEquals(Memo.class, list.get(0).getClass());
+      assertMemo1(list.get(0));
+
+      // get memo2 back from remote cache via query and check its attributes
+      query = qf.from(Memo.class)
+            .having("author.name").eq("Adrian")
+            .build();
+      list = query.list();
+      assertNotNull(list);
+//      assertEquals(1, list.size());
```

Comment:
There is a lot of code that is commented out in this file. Should they be added back in or removed?

Reply:
@anistor can you take care of this ?